cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -227,6 +234,9 @@ public ApplicationWithPrograms updateApp(Id.Application appId, AppRequest appReq
     if (currentSpec == null) {
       throw new ApplicationNotFoundException(appId);
     }
+    // App exists. Check if the current user has admin privileges on it before updating. The user's write privileges on
+    // the namespace will get enforced in the deployApp method.
+    authorizerInstantiatorService.get().enforce(appId.toEntityId(), SecurityRequestContext.toPrincipal(), Action.ADMIN);
```

Comment:
At the end of a successful deployment, he will have ALL access to the newly deployed app. Is that what you were asking?

Consider the following use case:
1. Brand new namespace *ns1*, Alice tries to deploy an app. It fails because she does not have WRITE access to *ns1*.
2. Alice is granted WRITE access to *ns1*. Now she can deploy an app there. She gets ALL privileges on the **app**
3. Now Bob wants to update (redeploy) the same app. Since it is an existing app, he needs ADMIN privileges on the **app**. 
4. If Bob wanted to deploy a new app, then he would need WRITE privileges on *ns1*